app/notification.go

```diff
@@ -534,12 +525,53 @@ type ExplicitMentions struct {
 	ChannelMentioned bool
 }
 
+const (
+	// Different types of mentions ordered by their priority from lowest to highest
+
+	// The post is in a thread that the user has commented on
+	ThreadMention MentionType = iota
```

Comment:
I would start with something like `NoMention` or `UnknownMention` (or even make it unexportable like `noMention`). With the rules below, we'll ovewrite it if there is any other mention, but if something goes wrong (bad initialization, for example) 0 would be a safe value that we can ignore most of the time.